src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1947,6 +1947,25 @@ def unregister_provider(cmd, resource_provider_namespace, wait=False):
     _update_provider(cmd.cli_ctx, resource_provider_namespace, registering=False, wait=wait)
 
 
+def register_management_group(cmd, resource_provider_namespace, management_group_id, wait=False):
+    import time
+    rcf = _resource_client_factory(cmd.cli_ctx)
+    r = rcf.providers.register_at_management_group_scope(resource_provider_namespace, management_group_id)
+    if r.registration_state == 'Registered':
```

Comment:
could you try to utilize the function_update_provider()?
https://github.com/Azure/azure-cli/blob/84eeeb0469054e2e054bc1cb234db371e433e590/src/azure-cli/azure/cli/command_modules/resource/custom.py#L963-L984